src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -220,6 +274,166 @@ private void triggerCheckpoint(String checkpointIdentifier) throws FlinkExceptio
         checkpointTrigger.triggerCheckpoint(checkpointId);
     }
 
+    // ------------------------------------------------------------------------
+    //  metrics
+    // ------------------------------------------------------------------------
+
+    /**
+     * Gauge for getting the unread bytes information from reader group.
+     */
+    private static class UnreadBytesGauge implements Gauge<Long> {
+
+        private final ReaderGroup readerGroup;
+
+        public UnreadBytesGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public Long getValue() {
+            return readerGroup.getMetrics().unreadBytes();
+        }
+    }
+
+    /**
+     * Gauge for getting the reader group name information from reader group.
+     */
+    private static class ReaderGroupNameGauge implements Gauge<String> {
+
+        private final ReaderGroup readerGroup;
+
+        public ReaderGroupNameGauge(ReaderGroup readerGroup) {
+            this.readerGroup = readerGroup;
+        }
+
+        @Override
+        public String getValue() {
+            return readerGroup.getGroupName();
+        }
+    }
+
+    /**
+     * Gauge for getting the scope name information from reader group.
+     */
+    private static class ScopeNameGauge implements Gauge<String> {
```

Comment:
I have kept the scope name metrics separate since a reader group scope could be different from the stream scope (multiple streams). We also report qualified stream name as another metric `StreamNamesGauge` which lists all the stream names as comma separated string.